### DISCUSSION TITLE
Can't create SDK from wsdl because it imports another WSDL document.

### DISCUSSION THREAD

**Discussion by OrakFR** [WsdlToPhp/PackageGenerator#290] [Q&A] [answered]
I need to creating a PHP system that consumes a SOAP web-service. There are several wsdl files, one for each endpoint. I was able to successfully generate PHP SDK for most of the wsdl files except for one that imports another wsdl file (a "Policies" wsdl file).

I have tried to "copy-paste" the "Policies" wsdl file in the wsdl endpoint, replacing the "import" line with no luck because when validating the final wsdl file in  [wsdl-analyzer.com](wsdl-analyzer.com) I got an "Error parsing your WSDL."

Is there any option to add this imported file url or any other solution?

Thanks in advance.

**Comment by mikaelcom** [comment]
Hi,
I must admit I don't know where the issue comes from in your case.
If you can share the WSDL causing issue publicly or via email at contact@wsdltophp.com, I'll try to find a way to make it work.

**Reply by OrakFR** [reply]
Sent you an email with the links to the wsdl files.
Thanks.

**Comment by mikaelcom** [comment, marked as answer]
Hi,
This should work by:
- downloading the root WSDL locally
- replacing `<wsp:PolicyReference URI="#SecureTransportBasicAuthPolicy" wsdl:required="true"/>` by `<wsp:PolicyReference URI="#SecureTransportBasicAuthPolicy" wsdl:required="false"/>` (`true` becomes `false`)

Enjoy (normally :smile:)

**Reply by OrakFR** [reply]
Hi, sorry for the delay.

Thanks for your answer... still haven't tested it yet. I'll keep you posted.

Thanx again.

**Reply by OrakFR** [reply]
Yup, that did the trick. Thanks.

**Comment by OrakFR** [comment]
Just one last question: I need to process multiple WSDL files using the PackageGenerator in order to obtain the complete PHP SDK. However, the issue I encountered is that the tool generates separate SDKs for each WSDL file, or at least that's how I understood it. Is there a way or a command to combine multiple WSDL files into a single PHP SDK?

Thanks once again.

**Reply by mikaelcom** [reply]
Normally, each WSDL corresponds to different methods wih different endpoints. But you can generate each SDK within a common main PHP namespace by specifying a namespace per SDK such as:
- Project\WebService1
- Project\WebService2
- Project\WebService2

Which will generate directories such as:
- src
  - Project
    - WebService1
    - WebService2

**Reply by OrakFR** [reply]
Cool. Thanks.
